### PR TITLE
Add remove button to book row

### PR DESCRIPTION
**Pull request by SilverMaiden**
This PR adds a delete button to each book row! This is hard deletion - we'll flesh out soft deletion later/better UX later.

Screenshot 2024-11-11 at 1 04 49 PM

### CODE REVIEW

File: src-ui/lib/components/BooksTable.svelte [hunk 1]
```diff
@@ -74,6 +79,7 @@
   <table class="table is-fullwidth">
     <thead>
       <tr>
+        <th></th>
```
Comment (rkofman) [inline src-ui/lib/components/BooksTable.svelte]: possible nit: add a comment for the delete action header.

File: src-ui/lib/components/BooksTable.svelte [hunk 2]
```diff
@@ -83,7 +89,7 @@
     </thead>
     <tbody>
       {#each booksStore.value as book}
-        <BooksTableRow {book} {handleEdit} />
+        <BooksTableRow {book} {handleEdit} {removeBook} />
```
Comment (rkofman) [inline src-ui/lib/components/BooksTable.svelte]: I wonder if `handleEdit` and `removeBook` shouldn't move inside of `BooksTableRow`? Part of the point of having a shared state for the `BooksStore` is that any component can use it directly and affect other components.

File: src-ui/lib/components/BooksTableRow.svelte [hunk 3]
```diff
@@ -3,19 +3,21 @@
 
   export let book: Book
   export let handleEdit: (book: Book, field: keyof Book, event: Event) => void
+  export let removeBook: (id: string) => Promise<void>
+  console.log('book is: ', book)
```
Comment (rkofman) [inline src-ui/lib/components/BooksTableRow.svelte]: remove `console.log`

File: src-ui/lib/components/BooksTableRow.svelte [hunk 4]
```diff
@@ -3,19 +3,21 @@
...
-  <td contenteditable="true" on:blur={(e) => handleEdit(book, 'title', e)}>{book.title || ''} </td>
-  <td contenteditable="true" on:blur={(e) => handleEdit(book, 'authors', e)}
-    >{book.authors?.join(', ') || ''}</td
-  >
-  <td contenteditable="true" on:blur={(e) => handleEdit(book, 'tags', e)}
-    >{book.tags?.join(', ') || ''}</td
-  >
+  <td contenteditable="true" on:blur={(e) => handleEdit(book, 'title', e)}>{book.title} </td>
+  <td contenteditable="true" on:blur={(e) => handleEdit(book, 'authors', e)}>{book.authors}</td>
+  <td contenteditable="true" on:blur={(e) => handleEdit(book, 'tags', e)}>{book.tags}</td>
```
Comment (rkofman) [inline src-ui/lib/components/BooksTableRow.svelte]: changes seem unrelated to book deletion?

**Review by rkofman — COMMENTED** [review]
Not sure if refactoring to move BoosStore interactions is worth it in this PR or a follow-up; but the other comments should def be addressed.

File: src-ui/lib/components/BooksTable.svelte [hunk 5]
```diff
@@ -74,6 +63,8 @@
   <table class="table is-fullwidth">
     <thead>
       <tr>
+        <!-- First blank header is for the "delete" button in each row -->
+        <th></th>
```
Comment (rkofman) [inline src-ui/lib/components/BooksTable.svelte]:
```suggestion
        <th><!-- delete action --></th>
```

**Review by rkofman — APPROVED** [review]

File: src-ui/lib/components/BooksTableRow.svelte [hunk 6]
```diff
@@ -1,15 +1,37 @@
...
 <tr>
+  <td
+    ><button aria-label="delete book" class="delete" on:click={() => removeBook(book.id)}
+    ></button></td
+  >
   <td contenteditable="true" on:blur={(e) => handleEdit(book, 'isbn10', e)}
     >{book.isbn10 ?? book.isbn13}</td
   >
-  <td contenteditable="true" on:blur={(e) => handleEdit(book, 'title', e)}>{book.title || ''} </td>
+  <td contenteditable="true" on:blur={(e) => handleEdit(book, 'title', e)}>{book?.title || ''} </td>
```
Comment (rkofman) [inline src-ui/lib/components/BooksTableRow.svelte]: why would book be `null`?

Comment (SilverMaiden) [reply src-ui/lib/components/BooksTableRow.svelte]: Left over from trying to do a bug fix, will remove!
